spot_driver_cpp/include/spot_driver_cpp/api/spot_api.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <spot_driver_cpp/api/image_client_api.hpp>
+#include <spot_driver_cpp/api/time_sync_api.hpp>
+#include <tl_expected/expected.hpp>
+
+#include <memory>
+#include <string>
+
+namespace spot_ros2 {
+
+class SpotApi {
+ public:
+  virtual tl::expected<void, std::string> createRobot(const std::string& ip_address, const std::string& robot_name) = 0;
```

Comment:
I suggest renaming this into 'initialize_robot', as it is not a factory anymore.